include/lte/lte_log.h

```diff
@@ -0,0 +1,187 @@
+/****************************************************************************
+ * apps/include/lte/lte_log.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __APPS_INCLUDE_LTE_LTE_LOG_H
+#define __APPS_INCLUDE_LTE_LTE_LOG_H
+
+/* API call type
+ *
+ * |     Sync API    |
+ * | --------------- |
+ * | lte_log_collect |
+ * | lte_log_getlist |
+ * | lte_log_open    |
+ * | lte_log_close   |
+ * | lte_log_read    |
+ * | lte_log_remove  |
+ * | lte_log_lseek   |
+ */
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/types.h>
+#include <nuttx/wireless/lte/lte.h>
+
+#ifdef __cplusplus
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/* Collect LTE modem FW logs and store them in the LTE modem storage.
+ *
+ * [out] output_fname: Buffer to store the name of the saved file.
+ *                     If the saved file name is not needed, set it to NULL.
+ * [in] len: Buffer length for output_fname. Sets LTE_LOG_NAME_LEN,
+ *           except when the saved file name is not needed.
+ *
+ * Return value : Returns 0 on success. On error, a negative value is
+ *                returned. The following values may be returned
+ *                in error cases.
+ *
+ * - ENOBUFS
+ * - ENOTSUP
+ *
+ */
+
+int lte_log_collect(char output_fname[], size_t len);
```

Comment:
Maybe can go with more traditional `char *output_fname`. I'm not sure if C89 will allow such constructions